### ISSUE TITLE
URI record type is incorrectly serialized/deserialized

### ISSUE THREAD

**Issue by Wolfizen** [PowerDNS/pdns#5443]
- Program: Authoritative, Recursor 
 - Issue type: Bug report

### Short description
PowerDNS incorrectly serializes/deserializes `URI` resource records. I have identified the cause of this issue, and linked a relevant portion of the source code.

### Environment
 - Operating system: Ubuntu Trusty (14.04) 64-bit
 - Software version: 4.0.3
 - Software source: PowerDNS repository `http://repo.powerdns.com/ubuntu trusty-rec-40 main`

### Steps to reproduce
1. Load the PDNS Recursor with a script similar to:
```lua
function preresolve(dq)
    dq.rcode = pdns.NOERROR
    dq:addAnswer(pdns.URI, '1 1 "http://example.com"', 300)
    return true
end
```
2. Query the recursor with `dig @[server] URI example.com`

### Expected behaviour
The answer section should include a URI record with the same `rdata` as given to `addAnswer`:
```
x.org. 300 IN URI 1 1 "http://example.com"
```

### Actual behaviour
The priority and weight values are wrong, and the URI is truncated by two characters.
```
; <<>> DiG 9.11.1 <<>> URI x.org @[server]
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 27571
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;x.org.				IN	URI

;; ANSWER SECTION:
x.org.			300	IN	URI	257 26740 "tp://example.com"

;; Query time: 0 msec
;; SERVER: [server]#53([server])
;; WHEN: Wed Jun 21 14:34:29 EDT 2017
;; MSG SIZE  rcvd: 68
```

### Other information
The root cause is that PDNS uses 8-bit integers for the Priority and Weight fields of the `URI` resource record, yet the standard is to use 16-bit integers. When PDNS computes the network data for this record, it pushes 8 bits for the priority, 8 bits for the weight, and variable-length for the uri. Yet when `dig` (or any other standard DNS client) attempts to parse this, it pops 16 bits for the priority (consuming both 8-bit numbers), 16 bits for the weight (consuming 2 characters of the uri) and variable-length for the uri (now missing two characters).

I have linked the (I presume) relevant PDNS source and RFC document specifying the URI record format.
RFC Document: https://tools.ietf.org/html/rfc7553#section-4.2
PDNS Source: https://github.com/PowerDNS/pdns/blob/749ac1478dd5f642f8859924201f809c06bd5f02/pdns/dnsrecords.cc#L472-L476

**Comment by Habbie** [comment]
Thank you for your extensive report! Please find the fix in #5444

**Comment by Wolfizen** [comment]
Thank you for the prompt fix! 👍
